ZenPacks/zenoss/Microsoft/Windows/migrate/RemoveWinRMServices.py

```diff
@@ -21,20 +21,30 @@
 # Zenoss Imports
 from Products.ZenModel.migrate.Migrate import Version
 from Products.ZenModel.ZenPack import ZenPackMigration
+from Products.Zuul.interfaces import ICatalogTool
 
 LOG = logging.getLogger('zen.MicrosoftWindows')
 
 
 class RemoveWinRMServices(ZenPackMigration):
-    version = Version(2, 7, 0)
+    version = Version(2, 7, 1)
 
     def migrate(self, pack):
-        org = pack.dmd.Devices.getOrganizer('/Server/Microsoft')
-        devices = org.getSubDevices()
-        device_count = len(devices)
-        if device_count:
-            LOG.info('Attempting to remove incompatible Windows Services from {} device{}.'
-                     .format(device_count, 's' if device_count > 1 else ''))
-            for device in devices:
-                device.os.removeRelation('winrmservices', suppress_events=True)
+
+        results = ICatalogTool(pack.getDmdRoot("Devices")).search(types=(
+            'ZenPacks.zenoss.Microsoft.Windows.BaseDevice.BaseDevice',
+        ))
+
+        if not results.total:
+            return
+
+        LOG.info('Attempting to remove incompatible Windows Services from {} device{}.'
+                     .format(results.total, 's' if results.total > 1 else ''))
+
+        for r in results:
+            try:
+                device = r.getObject()
                 device.os.buildRelations()
```

Comment:
This buildRelations() call is not necessary. It would be to handle the case of the winrmservices relationship not yet existing on device.os, but that case is already handled by the device.os.removeRelation call already being inside of an exception handler that ignores all exceptions.